Source/Core/Duality/Drawing/Canvas.cs

```diff
@@ -394,6 +394,168 @@ public void DrawThickLine(float x, float y, float x2, float y2, float width)
 			this.DrawThickLine(x, y, 0, x2, y2, 0, width);
 		}
 
+		/// <summary>
+		/// Draws an image.
+		/// </summary>
+		public void DrawImage(ContentRef<Material> material, float x, float y, float z, float w, float h)
+		{
+			this.PushState();
+
+			if (w < 0.0f) { x += w; w = -w; }
+			if (h < 0.0f) { y += h; h = -h; }
+
+			Vector3 pos = new Vector3(x, y, z);
+
+			Vector2 shapeHandle = pos.Xy;
+			float offset = this.State.DepthOffset;
+			ColorRgba shapeColor = material.Res.MainColor;
+			VertexC1P3T2[] vertices = this.RentVertices(4);
+
+			vertices[0].Pos = new Vector3(pos.X, pos.Y, pos.Z);
+			vertices[1].Pos = new Vector3(pos.X + w, pos.Y, pos.Z);
+			vertices[2].Pos = new Vector3(pos.X + w, pos.Y + h, pos.Z);
+			vertices[3].Pos = new Vector3(pos.X, pos.Y + h, pos.Z);
+
+			vertices[0].DepthOffset = offset;
+			vertices[1].DepthOffset = offset;
+			vertices[2].DepthOffset = offset;
+			vertices[3].DepthOffset = offset;
+
+			vertices[0].TexCoord = Vector2.Zero;
+			vertices[1].TexCoord = Vector2.UnitX;
+			vertices[2].TexCoord = Vector2.One;
+			vertices[3].TexCoord = Vector2.UnitY;
+
+			vertices[0].Color = shapeColor;
+			vertices[1].Color = shapeColor;
+			vertices[2].Color = shapeColor;
+			vertices[3].Color = shapeColor;
+
+			this.State.SetMaterial(material);
+			this.State.TransformVertices(vertices, shapeHandle);
+			this.device.AddVertices(this.State.MaterialDirect, VertexMode.LineLoop, vertices, 4);
```

Comment:
Why is this using `LineLoop`? Wouldn't that mean only drawing a one-pixel outline instead of the image?

Reply:
Aaand I apparently rolled back my commit.. first time I started from DrawRect by mistake and then something got messed up with my git repo. Thanks for the spot